plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2905,12 +2912,41 @@ public boolean delegateMaterializedViewRefreshToConnector(ConnectorSession sessi
     }
 
     @Override
-    public ConnectorInsertTableHandle beginRefreshMaterializedView(ConnectorSession session, ConnectorTableHandle tableHandle, List<ConnectorTableHandle> sourceTableHandles, RetryMode retryMode)
+    public ConnectorInsertTableHandle beginRefreshMaterializedView(ConnectorSession session, ConnectorTableHandle tableHandle, List<ConnectorTableHandle> sourceTableHandles, RetryMode retryMode, RefreshType refreshType)
     {
+        checkState(fromSnapshotsForRefresh.isEmpty(), "Snapshot map must be empty at the start of MV refresh operation.");
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
         beginTransaction(icebergTable);
 
+        Optional<String> dependencies = Optional.ofNullable(icebergTable.currentSnapshot())
+                .map(Snapshot::summary)
+                .map(summary -> summary.get(DEPENDS_ON_TABLES));
+
+        boolean shouldUseIncremental = isIncrementalRefreshEnabled(session)
+                && refreshType == RefreshType.INCREMENTAL
+                // there is a single source table
+                && sourceTableHandles.size() == 1
+                // and source table is an Iceberg table
+                && getOnlyElement(sourceTableHandles) instanceof IcebergTableHandle
+                // and the source table's fromSnapshot is available in the MV snapshot summary
+                && dependencies.isPresent() && !dependencies.get().equals(UNKNOWN_SNAPSHOT_TOKEN);
+
+        if (shouldUseIncremental) {
+            Map<String, String> baseTableToSnapshot = MAP_SPLITTER.split(dependencies.get());
```

Comment:
now that we assume only single table this can be simplified:
```
Map<String, String> baseTableToSnapshot = MAP_SPLITTER.split(dependencies.get());
checkState(baseTableToSnapshot.size() == 1, "Expected %s to contain only single element", baseTableToSnapshot);
Map.Entry<String, String> baseTable = getOnlyElement(baseTableToSnapshot.entrySet());
String[] schemaTable = baseTable.getKey().split("\\.");
baseSchemaTable = new SchemaTableName(schemaTable[0], schemaTable[1]);
checkState(baseSchemaTable.equals(handle.getSchemaTableName()), "Base table name %s doesn't match source table name %s, baseSchemaTable, handle.getSchemaTableName())
fromSnapshotsForRefresh = Optional.of(Long.parseLong(baseTable.getValue())));
```